### ISSUE TITLE
wflow_git_push with personal access token

### ISSUE THREAD

**Issue by pcarbo** [workflowr/workflowr#272]
@jdblischak @stephens999 I was able to successfully complete [all the steps in the tutorial](https://workflowr.github.io/workflowr/articles/wflow-01-getting-started.html) after creating a [personal access token (PAT)](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token):

```r
library(workflowr)
wflow_start()
wflow_start("myproject2")
wflow_build()
wflow_publish(c("analysis/index.Rmd", "analysis/about.Rmd", "analysis/license.Rmd"),
  "Publish the initial files for myproject")
wflow_use_github("pcarbo")
wflow_git_push()
# Please enter your username (Esc to cancel): pcarbo
# Please enter your password: ****************************************
# Summary from wflow_git_push
# Pushing to the branch "master" of the remote repository "origin"
# Using the HTTPS protocol
# The following Git command was run:
#  $ git push origin master
```

Providing the PAT at the password prompt worked.

However, the PAT needs to be generated with the right OAuth scope for this to work, and I'm not sure (yet) what exact scope is needed. I can say that setting the scope to this worked for me:

<img width="538" alt="scope" src="https://user-images.githubusercontent.com/1503904/166938161-1b0ea83d-3c7a-40be-be74-b69153ddcfba.png">

Overall, this approach is more annoying because you have to keep track of your PAT and copy and paste it into the password prompt each time, but at least it works.

**Comment by pcarbo** [comment]
There is also the potential to use [git2r::cred_token](https://docs.ropensci.org/git2r/reference/cred_token.html) to automate this for expert workflowr users, but we'd have to figure out how to integrate this into the existing wflow_start interface.

**Comment by jdblischak** [comment]
> Overall, this approach is more annoying because you have to keep track of your PAT and copy and paste it into the password prompt each time, but at least it works.

Yes, it works, but it is super tedious. Copy-pasting a PAT repeatedly is too cumbersome to be viable long term for a user, which is why I haven't bothered documenting it (since that would be lengthy). With the current GitHub requirements, it is easier to setup SSH keys for authentication.

Long-term for workflowr we need to use git2r/gert/credentials to properly and securely cache the user's PAT for authentication. There was an initial effort to integrate credentials into git2r, but it has stalled https://github.com/ropensci/git2r/pull/422

That being said, `git2r::cred_token()` is probably the most straightforward to document for end users and to integrate into  `wflow_git_push()` and `wflow_git_pull()`.

**Comment by jdblischak** [comment]
Also see #235
